operator/standalone/pipeline.go

```diff
@@ -2,26 +2,201 @@ package standalone
 )
 
-type clientKey struct{}
+// fetchHelmRelease fetches the Helm release for the given instance.
+func fetchHelmRelease(ctx context.Context) error {
+	helmRelease := &helmv1beta1.Release{}
+	err := getClientFromContext(ctx).Get(ctx, client.ObjectKey{Name: getInstanceFromContext(ctx).Status.HelmChart.DeploymentNamespace}, helmRelease)
+	setHelmReleaseInContext(ctx, helmRelease)
+	return err
+}
+
+// fetchOperatorConfigF fetches a matching v1alpha1.PostgresqlStandaloneOperatorConfig from the OperatorNamespace.
+// The Major version specified in v1alpha1.PostgresqlStandalone is used to filter the correct config by the v1alpha1.PostgresqlMajorVersionLabelKey label.
+// If there is none or multiple found, it returns an error.
+func fetchOperatorConfigF(operatorNamespace string) func(ctx2 context.Context) error {
+	return func(ctx context.Context) error {
+		list := &v1alpha1.PostgresqlStandaloneOperatorConfigList{}
+		labelMatch := client.MatchingLabels{
+			v1alpha1.PostgresqlMajorVersionLabelKey: getInstanceFromContext(ctx).Spec.Parameters.MajorVersion.String(),
+		}
+		ns := client.InNamespace(operatorNamespace)
+		err := getClientFromContext(ctx).List(ctx, list, labelMatch, ns)
+		if err != nil {
+			return err
+		}
+		if len(list.Items) == 0 {
+			return fmt.Errorf("no %s found with label '%s' in namespace '%s'", v1alpha1.PostgresqlStandaloneOperatorConfigKind, labelMatch, ns)
+		}
+		if len(list.Items) > 1 {
+			return fmt.Errorf("multiple versions of %s found with label '%s' in namespace '%s'", v1alpha1.PostgresqlStandaloneOperatorConfigKind, labelMatch, ns)
+		}
+		setConfigInContext(ctx, &list.Items[0])
+		return nil
+	}
+}
 
-func setClientInContext(ctx context.Context, c client.Client) {
-	pipeline.StoreInContext(ctx, clientKey{}, c)
+// applyValuesFromInstance merges the user-defined and -exposed Helm values into the current Helm values map.
+func applyValuesFromInstance(ctx context.Context) error {
+	instance := getInstanceFromContext(ctx)
+	resources := helmvalues.V{
+		"auth": helmvalues.V{
+			"enablePostgresUser": true, // See https://github.com/vshn/appcat-service-postgresql/issues/83 why we always create a superuser
+			"existingSecret":     getCredentialSecretName(),
+			"database":           instance.Name,
+			"username":           instance.Name,
+		},
+		"primary": helmvalues.V{
+			"resources": helmvalues.V{
+				"limits": helmvalues.V{
+					"memory": instance.Spec.Parameters.Resources.MemoryLimit.String(),
+				},
+			},
+			"persistence": helmvalues.V{
+				"existingClaim": getPVCName(),
+			},
+			"podAnnotations": helmvalues.V{ // these annotations can stay, even if backups are disabled.
+				"k8up.io/backupcommand":                      `sh -c 'PGUSER="postgres" PGPASSWORD="$POSTGRES_POSTGRES_PASSWORD" pg_dumpall --clean'`,
+				"k8up.io/file-extension":                     ".sql",
+				"postgresql.appcat.vshn.io/storage-capacity": instance.Spec.Parameters.Resources.StorageCapacity.String(),
```

Comment:
`Spec.Parameters.Resources.StorageCapacity` is technically and currently not required. But there's also no default being set in the webhook either.
If this field is not set, this will lead to a nil panic. Basically we need _some_ value for the storage. Question is, from where? (hardcoded default if not set, or required by the spec/webhook)

Reply:
We could try to borrow the default value from [Bitnami Chart](https://github.com/bitnami/charts/blob/master/bitnami/postgresql/values.yaml#L657)

Reply:
So in essence you'd set a default?

Right now, I'm on the side to make it a required field. That way, the users have to think about the capacity and aren't surprised if they find that the instance is full. For example, if they only apply the minimal YAML coming from a tutorial or so, they might not know that their instance is only `x` Gi of capacity by default and then full. They might be under assumption that this scales as needed, like in AWS RDS for instance. Basically defaulting to some intransparent value might set wrong expectations regarding the capacity.